src/Components/ProvisioningWizard/steps/AccountCustomizations/index.js

```diff
@@ -0,0 +1,39 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { AWS_PROVIDER, GCP_PROVIDER } from '../../../Common/constants';
+import { defaultRegionByProvider } from '../../../Common/helpers';
+import { useWizardContext } from '../../../Common/WizardContext';
+import AWS from './aws';
+import GCP from './gcp';
+
+const AccountCustomizations = ({ setStepValidated, architecture, composeID, provider }) => {
+  const [, setWizardContext] = useWizardContext();
+
+  React.useEffect(() => {
+    provider &&
+      setWizardContext((prevState) => ({
+        ...prevState,
+        provider,
+        chosenRegion: defaultRegionByProvider(provider),
+        chosenImageID: composeID,
+      }));
+  }, [provider, setWizardContext]);
```

Comment:
This feels bit out of place, shouldn't we rather set the provider in the parent component (`ProvisioningWizard`) and only use it from the context here?